airbyte-cdk/java/airbyte-cdk/core/src/main/kotlin/io/airbyte/cdk/core/IntegrationCommand.kt

```diff
@@ -0,0 +1,165 @@
+                "\t read - reads the source and outputs messages to STDOUT",
+                "\t spec - outputs the json configuration specification",
+                "\t write - writes messages from STDIN to the integration",
+            ],
+    )
+    lateinit var command: String
+
+    @CommandLine.Option(
+        names = ["--" + JavaBaseConstants.ARGS_CONFIG_KEY],
+        description =
+            [
+                JavaBaseConstants.ARGS_CONFIG_DESC,
+                "Required by the following commands: check, discover, read, write",
+            ],
+    )
+    lateinit var configFile: Path
+
+    @CommandLine.Option(
+        names = ["--" + JavaBaseConstants.ARGS_CATALOG_KEY],
+        description =
+            [
+                JavaBaseConstants.ARGS_CATALOG_DESC,
+                "Required by the following commands: read, write",
+            ],
+    )
+    lateinit var catalogFile: Path
+
+    @CommandLine.Option(
+        names = ["--" + JavaBaseConstants.ARGS_STATE_KEY],
+        description =
+            [
+                JavaBaseConstants.ARGS_PATH_DESC,
+                "Required by the following commands: read",
+            ],
+    )
+    lateinit var stateFile: Optional<Path>
+
+    @CommandLine.Spec lateinit var commandSpec: CommandLine.Model.CommandSpec
+
+    override fun run() {
+        val result = execute(command)
+        result.onSuccess { airbyteMessage ->
+            airbyteMessage?.let { outputRecordCollector.accept(airbyteMessage) }
+        }
+        result.onFailure {
+            commandSpec.commandLine().usage(System.out)
+            logger.error(it) { "\nUnable to perform operation '$command'." }
+            // Many of the exceptions thrown are nested inside layers of RuntimeExceptions. An
+            // attempt is made
+            // to
+            // find the root exception that corresponds to a configuration error. If that does not
+            // exist, we
+            // just return the original exception.
+            ApmTraceUtils.addExceptionToTrace(it)
+            val rootThrowable = ConnectorExceptionUtil.getRootConfigError(Exception(it))
+            val displayMessage = ConnectorExceptionUtil.getDisplayMessage(rootThrowable)
+            // If the source connector throws a config error, a trace message with the relevant
+            // message should
+            // be surfaced.
+            if (ConnectorExceptionUtil.isConfigError(rootThrowable)) {
+                AirbyteTraceMessageUtility.emitConfigErrorTrace(it, displayMessage)
+            }
+            if (OperationType.CHECK.name.equals(command, true)) {
+                // Currently, special handling is required for the CHECK case since the user display
+                // information in
+                // the trace message is
+                // not properly surfaced to the FE. In the future, we can remove this and just throw
+                // an exception.
```

Comment:
Is there an issue on github tracking this?

Reply:
@postamar Don't know.  This comment is from the `IntegrationRunner` code and has existed since at least 11/22.

Reply:
Ooof. How would you feel with removing this, and let's put it back later on in if we need to? This is our chance to deal with warts like these.

Reply:
Similar sentiment of another comment - lets make backlog tickets for anything not directly related to the intent of this PR and not block merging on logic that has existed.

Reply:
Sounds good to me.